### ISSUE TITLE
Kafka TLS for load balancer type is only allowing the elb dns address, not the override advertisedHosts

### ISSUE THREAD

**Issue by georambo** [strimzi/strimzi-kafka-operator#3330]
**Describe the bug**
When I create a Kafka object, with external listeners, and overrides for the advertised host, with tls enabled, the TLS certificate created only has an allowed address of the AWS elb url.  The advertised host from the override is not listed as an address for the certificate for brokers, or bootstrap servers.

slack thread to conversation on CNCF slack: https://cloud-native.slack.com/archives/CMH3Q3SNP/p1594897537105300

**To Reproduce**
2. Create Custom Resource see yaml below
3. Describe the Kafka object.  
```
kubectl describe Kafka my-cluster
```

**Expected behavior**
I expect the TLS certificate for the External Load Balancers to include my advertised host overrides, so that I can use TLS at my custom domain.

**Environment (please complete the following information):**
 - Strimzi version: 0.18.0
 - Installation method: YAML files
 - Kubernetes cluster: EKS v1.16.8-eks-e16311
 - Infrastructure: Amazon EKS

**YAML files and logs**
```
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: my-cluster
  namespace: kafka
spec:
  kafka:
    tolerations:
      - key: "lifecycle"
        operator: "Equal"
        value: "Ec2Spot"
        effect: "NoSchedule"
    version: 2.5.0
    replicas: 3
    resources:
      requests:
        memory: 3Gi
        cpu: "2"
      limits:
        memory: 8Gi
        cpu: "4"
    jvmOptions:
      -Xms: 3000m
      -Xmx: 8192m
    listeners:
      external:
        type: loadbalancer
        tls: false
        overrides:
          bootstrap:
            dnsAnnotations:
              external-dns.alpha.kubernetes.io/hostname: kafka-bootstrap.test.plc
              external-dns.alpha.kubernetes.io/sync: "true"
              external-dns.alpha.kubernetes.io/ttl: "60"
            address: kafka-bootstrap.test.plc
          brokers:
          - broker: 0
            dnsAnnotations:
              external-dns.alpha.kubernetes.io/hostname: kafka-broker-0.test.plc
              external-dns.alpha.kubernetes.io/sync: "true"
              external-dns.alpha.kubernetes.io/ttl: "60"
            advertisedHost: kafka-broker-0.test.plc
          - broker: 1
            dnsAnnotations:
              external-dns.alpha.kubernetes.io/hostname: kafka-broker-1.test.plc
              external-dns.alpha.kubernetes.io/sync: "true"
              external-dns.alpha.kubernetes.io/ttl: "60"
            advertisedHost: kafka-broker-1.test.plc
          - broker: 2
            dnsAnnotations:
              external-dns.alpha.kubernetes.io/hostname: kafka-broker-2.test.plc
              external-dns.alpha.kubernetes.io/sync: "true"
              external-dns.alpha.kubernetes.io/ttl: "60"
            advertisedHost: kafka-broker-2.test.plc
      plain: {}
      tls: {}
    template:
      externalBootstrapService:
        loadBalancerSourceRanges:
          - 192.168.0.0/18
          - 100.64.0.0/10
        metadata:
          annotations:
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled: "true"
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-timeout: "60"
            service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
      perPodService:
        loadBalancerSourceRanges:
          - 192.168.0.0/18
          - 100.64.0.0/10
        metadata:
          annotations:
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled: "true"
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-timeout: "60"
            service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.5"
      auto.create.topics.enable: false
      delete.topic.enable: true
      group.initial.rebalance.delay.ms: 0
      log.retention.check.interval.ms: 300000
      log.retention.hours: 168
      log.segment.bytes: 1073741824
      num.io.threads: 8
      num.network.threads: 3
      num.partitions: 1
      num.recovery.threads.per.data.dir: 1
      socket.receive.buffer.bytes: 102400
      socket.request.max.bytes: 104857600
      socket.send.buffer.bytes: 102400
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 200Gi
        deleteClaim: false
  zookeeper:
    tolerations:
      - key: "lifecycle"
        operator: "Equal"
        value: "Ec2Spot"
        effect: "NoSchedule"
    replicas: 3
    resources:
      requests:
        memory: 3Gi
        cpu: "2"
      limits:
        memory: 8Gi
        cpu: "4"
    jvmOptions:
      -Xms: 3096m
      -Xmx: 8192m
    storage:
      type: persistent-claim
      size: 200Gi
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}
  cruiseControl: {}

```

`kubectl describe Kafka my-cluster`
```
Name:         my-cluster
Namespace:    kafka
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"kafka.strimzi.io/v1beta1","kind":"Kafka","metadata":{"annotations":{},"name":"my-cluster","namespace":"kafka"},"spec":{"cruiseCont...
API Version:  kafka.strimzi.io/v1beta1
Kind:         Kafka
Metadata:
  Creation Timestamp:  2020-07-07T19:13:33Z
  Generation:          3
  Resource Version:    66668408
  Self Link:           /apis/kafka.strimzi.io/v1beta1/namespaces/kafka/kafkas/gpd
  UID:                 2d52d2bf-6803-407b-9763-e7d095e97256
Spec:
  Cruise Control:
  Entity Operator:
    Topic Operator:
    User Operator:
  Kafka:
    Config:
      auto.create.topics.enable:                 false
      delete.topic.enable:                       true
      group.initial.rebalance.delay.ms:          0
      log.message.format.version:                2.5
      log.retention.check.interval.ms:           300000
      log.retention.hours:                       168
      log.segment.bytes:                         1073741824
      num.io.threads:                            8
      num.network.threads:                       3
      num.partitions:                            1
      num.recovery.threads.per.data.dir:         1
      offsets.topic.replication.factor:          1
      socket.receive.buffer.bytes:               102400
      socket.request.max.bytes:                  104857600
      socket.send.buffer.bytes:                  102400
      transaction.state.log.min.isr:             2
      transaction.state.log.replication.factor:  3
    Jvm Options:
      - Xms:  3000m
      - Xmx:  8192m
    Listeners:
      External:
        Overrides:
          Bootstrap:
            Address:  kafka-bootstrap.test.com
            Dns Annotations:
              external-dns.alpha.kubernetes.io/hostname:  kafka-bootstrap.test.com
              external-dns.alpha.kubernetes.io/sync:      true
              external-dns.alpha.kubernetes.io/ttl:       60
          Brokers:
            Advertised Host:  kafka-broker-0.test.com
            Broker:           0
            Dns Annotations:
              external-dns.alpha.kubernetes.io/hostname:  kafka-broker-0.test.com
              external-dns.alpha.kubernetes.io/sync:      true
              external-dns.alpha.kubernetes.io/ttl:       60
            Advertised Host:                              kafka-broker-1.test.com
            Broker:                                       1
            Dns Annotations:
              external-dns.alpha.kubernetes.io/hostname:  kafka-broker-1.test.com
              external-dns.alpha.kubernetes.io/sync:      true
              external-dns.alpha.kubernetes.io/ttl:       60
            Advertised Host:                              kafka-broker-2.test.com
            Broker:                                       2
            Dns Annotations:
              external-dns.alpha.kubernetes.io/hostname:  kafka-broker-2.test.com
              external-dns.alpha.kubernetes.io/sync:      true
              external-dns.alpha.kubernetes.io/ttl:       60
        Type:                                             loadbalancer
      Plain:
      Tls:
    Replicas:  3
    Resources:
      Limits:
        Cpu:     4
        Memory:  8Gi
      Requests:
        Cpu:     2
        Memory:  3Gi
    Storage:
      Type:  jbod
      Volumes:
        Delete Claim:  false
        Id:            0
        Size:          200Gi
        Type:          persistent-claim
    Template:
      External Bootstrap Service:
        Metadata:
          Annotations:
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled:  true
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-timeout:  60
            service.beta.kubernetes.io/aws-load-balancer-internal:                     0.0.0.0/0
      Per Pod Service:
        Metadata:
          Annotations:
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled:  true
            service.beta.kubernetes.io/aws-load-balancer-connection-draining-timeout:  60
            service.beta.kubernetes.io/aws-load-balancer-internal:                     0.0.0.0/0
    Tolerations:
      Effect:    NoSchedule
      Key:       lifecycle
      Operator:  Equal
      Value:     Ec2Spot
    Version:     2.5.0
  Zookeeper:
    Jvm Options:
      - Xms:   3096m
      - Xmx:   8192m
    Replicas:  3
    Resources:
      Limits:
        Cpu:     4
        Memory:  8Gi
      Requests:
        Cpu:     2
        Memory:  3Gi
    Storage:
      Delete Claim:  false
      Size:          200Gi
      Type:          persistent-claim
    Tolerations:
      Effect:    NoSchedule
      Key:       lifecycle
      Operator:  Equal
      Value:     Ec2Spot
Status:
  Conditions:
    Last Transition Time:  2020-07-16T10:44:53+0000
    Status:                True
    Type:                  Ready
  Listeners:
    Addresses:
      Host:             gpd-kafka-bootstrap.kafka.svc
      Port:             9092
    Bootstrap Servers:  gpd-kafka-bootstrap.kafka.svc:9092
    Type:               plain
    Addresses:
      Host:             gpd-kafka-bootstrap.kafka.svc
      Port:             9093
    Bootstrap Servers:  gpd-kafka-bootstrap.kafka.svc:9093
    Certificates:
      -----BEGIN CERTIFICATE-----
<redacted>
-----END CERTIFICATE-----
    Type:  tls
    Addresses:
      Host:             internal-xxxxxx.us-east-1.elb.amazonaws.com
      Port:             9094
    Bootstrap Servers:  internal-xxxxx.us-east-1.elb.amazonaws.com:9094
    Certificates:
      -----BEGIN CERTIFICATE-----
<redacted>
-----END CERTIFICATE-----
    Type:               external
  Observed Generation:  3
Events:                 <none>
```

**Comment by scholzj** [comment]
It indeed seems that the address in the `.status` is wrong. However, when I decode the certificate, it looks OK to me with the following list of SANs:

* my-cluster-kafka-bootstrap.myproject.svc
* my-cluster-kafka-bootstrap.myproject.svc.cluster.local
* my-cluster-kafka-bootstrap
* my-cluster-kafka-bootstrap.myproject
* my-cluster-kafka-brokers.myproject.svc
* my-cluster-kafka-brokers.myproject.svc.cluster.local
* my-cluster-kafka-brokers
* my-cluster-kafka-brokers.myproject
* af3d8e95a9e0b407aaf6b9c4cffb16e2-89600791.eu-west-1.elb.amazonaws.com
* my-cluster-kafka-1.my-cluster-kafka-brokers.myproject.svc.cluster.local
* kafka-broker-1.test.plc
* kafka-bootstrap.test.plc
* a7095c2dd9c0a4bf9a15037ef9501c27-504724111.eu-west-1.elb.amazonaws.com
* my-cluster-kafka-1.my-cluster-kafka-brokers.myproject.svc

which contains both the ELB as well as the custom address. So that part seems to be ok for me.

**Comment by scholzj** [comment]
This should be fixed now in Strimzi 0.20.0. Please have a look and give it a try.
